src/librustc/ty/sty.rs

```diff
@@ -1504,6 +1504,50 @@ impl<'a, 'gcx, 'tcx> TyS<'tcx> {
         }
     }
 
+    /// Checks whether a type is definitely uninhabited. This is
+    /// conservative: for some types that are uninhabited we return `false`,
+    /// but we only return `true` for types that are definitely uninhabited.
+    /// `ty.conservative_is_uninhabited` implies that any value of type `ty`
+    /// will be `Abi::Uninhabited`.
```

Comment:
It is perhaps worth noting that **uninhabited types can still have non-zero size** (e.g., `(u32, !)`) due to the possibility of uninitialized data.

Reply:
maybe add a comment here?